reactor-core/src/main/java/reactor/core/publisher/FluxBufferPredicate.java

```diff
@@ -156,11 +164,11 @@ public void request(long n) {
 					// we'll continue requesting one by one)
 					if (!DrainUtils.postCompleteRequest(n,
 							actual,
-							this,
-							REQUESTED,
+							this, REQUESTED_BUFFERS,
 							this,
 							this)) {
-						s.request(1);
+						REQUESTED_FROM_SOURCE.addAndGet(this, n);
```

Comment:
should not it be added by `Operators#addCap()` ?

Reply:
indeed better to use capped add 👍